### PR TITLE
1276 - Reduced the memory foot print of the email parser

### PR DESCRIPTION
**Pull request by kellykelly3**

### CODE REVIEW

File: thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/MboxParser.java [hunk 1]
```diff
@@ -97,63 +95,76 @@
...
-        Iterable<CharBufferWrapper> mboxIterator = null;
+        
         // Loop through the possible encoders and find the first one that works.
         // That will usually be one of the first ones.
         for (CharsetEncoder encoder : encoders) {
             try {
-                mboxIterator = MboxIterator.fromFile(mboxFile).charset(encoder.charset()).build();
-                theEncoder = encoder;
+                Iterable<CharBufferWrapper> mboxIterable = MboxIterator.fromFile(mboxFile).charset(encoder.charset()).build();
+                if(mboxIterable != null) {
```
Comment (raman-bt) [inline thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/MboxParser.java]: I prefer a whitespace between the 'if' and the opening '('.  I think most of our existing code follows that convention.

I'm not sure if our coding guidelines specify it explicitly. 

There's many instances of "if(" and a few of "while(" in this PR.  I'll let @rcordovano decide if they should be changed in this PR and provide guidance of future.

File: thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/PstParser.java [hunk 2]
```diff
@@ -55,112 +57,193 @@
...
     String getErrors() {
         return errors.toString();
     }
+    
+    int getFailureCount() {
+        return failureCount;
+    }
+     
+     /**
+     * Get an Iterator to which will iterate over the PSTFile, but return EmailMessages
```
[the remark this answers is not in the document]
Comment (kellykelly3) [reply thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/PstParser.java]: "Get an Iterator which will"  the 'to' is erroneous.

File: thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/MboxParser.java (hunk 1 above)
Comment (kellykelly3) [reply thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/MboxParser.java]: I'll just fix it.

File: thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/PstParser.java [hunk 3]
```diff
@@ -55,112 +57,193 @@
...
+            while(iterator.hasNext()) {
+                messages.add(iterator.next());
+            }
+        }
+        
+        return messages;
     }
-
+    
+    /**
```
[the remark this answers is not in the document]
Comment (kellykelly3) [reply thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/PstParser.java]: because I forgot.

**Comment by rcordovano** [comment]
@kellykelly3, please resolve the Codacy issues (except the complexity issue; we will deal wit that at a later time), thanks!

**Comment by kellykelly3** [comment]
Codacy issues addressed.

**Comment by raman-bt** [comment]
@kellykelly3: could you resolve the merge conflict. I'll give it another round of test then.

**Comment by rcordovano** [comment]
@kellykelly3, let's address the Codacy issues. Skimming the code, I think that one approach would be to have a collection of error strings instead of a single string built up with multiple concatenations.

**Comment by rcordovano** [comment]
@kellykelly3, now there are two more quick Codacy issue fixes to do, thanks.

**Pull request merged**